account_cutoff_base/models/account_cutoff.py

```diff
@@ -0,0 +1,391 @@
+        to_provision = {}
+        merge_keys = self._get_merge_keys()
+        for provision_line in provision_lines:
+            key = tuple([provision_line.get(key) for key in merge_keys])
+            if key in to_provision:
+                to_provision[key] += provision_line['amount']
+            else:
+                to_provision[key] = provision_line['amount']
+        return to_provision
+
+    @api.multi
+    def create_move(self):
+        self.ensure_one()
+        move_obj = self.env['account.move']
+        if self.move_id:
+            raise UserError(_(
+                "The Cut-off Journal Entry already exists. You should "
+                "delete it before running this function."))
+        if not self.line_ids:
+            raise UserError(_(
+                "There are no lines on this Cut-off, so we can't create "
+                "a Journal Entry."))
+        provision_lines = []
+        for line in self.line_ids:
+            provision_lines.append(
+                self._prepare_provision_line(line))
+            for tax_line in line.tax_line_ids:
+                provision_lines.append(
+                    self._prepare_provision_tax_line(tax_line))
+        to_provision = self._merge_provision_lines(provision_lines)
+        vals = self._prepare_move(to_provision)
+        move = move_obj.create(vals)
+        self.write({'move_id': move.id, 'state': 'done'})
+
+        action = self.env['ir.actions.act_window'].for_xml_id(
+            'account', 'action_move_journal_line')
+        action.update({
+            'view_mode': 'form,tree',
+            'res_id': move.id,
+            'view_id': False,
+            'views': False,
+        })
+        return action
+
+
+class AccountCutoffLine(models.Model):
+    _name = 'account.cutoff.line'
+    _description = 'Account Cut-off Line'
+
+    parent_id = fields.Many2one(
+        'account.cutoff', string='Cut-off', ondelete='cascade')
+    name = fields.Char('Description')
+    company_currency_id = fields.Many2one(
+        related='parent_id.company_currency_id',
+        string="Company Currency", readonly=True)
+    partner_id = fields.Many2one(
+        'res.partner', string='Partner', readonly=True)
+    account_id = fields.Many2one(
+        'account.account', 'Account',
+        domain=[('deprecated', '=', False)], required=True, readonly=True)
+    cutoff_account_id = fields.Many2one(
+        'account.account', string='Cut-off Account',
+        domain=[('deprecated', '=', False)], required=True, readonly=True)
+    cutoff_account_code = fields.Char(
+        related='cutoff_account_id.code',
+        string='Cut-off Account Code', readonly=True)
+    analytic_account_id = fields.Many2one(
+        'account.analytic.account', string='Analytic Account',
+        domain=[('account_type', '!=', 'closed')], readonly=True)
+    analytic_account_code = fields.Char(
+        related='analytic_account_id.code',
+        string='Analytic Account Code', readonly=True)
```

Comment:
The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
        readonly=True)
```